sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -393,6 +393,21 @@ case class PreprocessTableInsertion(conf: SQLConf) extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * SPARK-28313: Spark sql null type incompatible with hive void type
+ */
+object CreateTableCheck extends Rule[LogicalPlan] {
```

Comment:
Can we make a new PR for this change? This is a behavior change, which needs more discussion and needs to add something to migration guide.

Reply:
Sure. As i say, this pr has two goals.
1. compatibility with hive
2. reject create table use NullType
So it's ok to make new pr for goal 2.